gcp/workers/linter/linter.py

```diff
@@ -147,6 +149,38 @@ def record_quality_finding(bug_id: str, source: str,
                   new_findings, source)
 
 
+def upload_record_to_bucket(json_result: Any):
+  """Uploads the linter result to a GCS bucket, creating one file per source."""
+  storage_client = storage.Client()
+  bucket = storage_client.get_bucket(LINTER_EXPORT_BUCKET)
+  prefix_to_source = construct_prefix_to_source_map()
+
+  source_results = {}
+  for filename, findings in json_result.items():
+    bug_id = os.path.splitext(os.path.basename(filename))[0]
+    prefix = bug_id.split('-')[0] + '-'
+    source = prefix_to_source.get(prefix, '')
+
+    if source not in source_results:
+      source_results[source] = {}
+    source_results[source][filename] = findings
+
+  logging.info('Uploading linter results for %d sources.', len(source_results))
+  for source, results in source_results.items():
+    if not results:
+      continue
+
+    target_path = os.path.join(LINTER_RESULT_DIR, source, 'result.json')
+    logging.info('Uploading linter result for source '
+                 '%s'
+                 ' to %s/%s', source, LINTER_EXPORT_BUCKET, target_path)
+
+    blob = bucket.blob(target_path)
+    blob.upload_from_string(
```

Comment:
Should we be cleaning up previous uploads? E.g. for records that have been fixed.

Reply:
Depends on if the bucket is the source of truth, or you are expecting everyone to go through the API.

Reply:
Good catch! Added cleanup function before writing new ones.